portality/tasks/article_bulk_delete.py

```diff
@@ -0,0 +1,128 @@
+from copy import deepcopy
+import json
+
+from flask_login import current_user
+
+from portality import models
+
+from portality.tasks.redis_huey import main_queue
+from portality.decorators import write_required
+
+from portality.background import AdminBackgroundTask, BackgroundApi, BackgroundException
+
+
+def article_bulk_delete_manage(selection_query, dry_run=True):
+
+    if dry_run:
+        ArticleBulkDeleteBackgroundTask.check_admin_privilege(current_user.id)
+        return ArticleBulkDeleteBackgroundTask.estimate_delete_counts(selection_query)
+
+    ids = ArticleBulkDeleteBackgroundTask.resolve_selection_query(selection_query)
+    job = ArticleBulkDeleteBackgroundTask.prepare(
+        current_user.id,
+        selection_query=selection_query,
+        ids=ids
+    )
+    ArticleBulkDeleteBackgroundTask.submit(job)
+
+    return True
+
+
+class ArticleBulkDeleteBackgroundTask(AdminBackgroundTask):
+
+    __action__ = "article_bulk_delete"
+
+    @classmethod
+    def _job_parameter_check(cls, params):
+        # we definitely need "ids" defined
+        return bool(cls.get_param(params, 'ids'))
+
+    def run(self):
+        """
+        Execute the task as specified by the background_job
+        :return:
+        """
+        job = self.background_job
+        params = job.params
+
+        ids = self.get_param(params, 'ids')
+
+        if not self._job_parameter_check(params):
+            raise BackgroundException(u"{}.run run without sufficient parameters".format(self.__class__.__name__))
+
+        estimate = self.estimate_delete_counts(json.loads(job.reference['selection_query']))
+        job.add_audit_message(u"About to delete an estimated {} articles".format(estimate))
+
+        # TODO this could be an issue with lots of article IDs (the should_terms being so big)
+        # Break up in batches if a problem occurs during manual testing.
+        # Might need a unit or integration test to test batching up works.
+        article_delete_q_by_ids = models.Article.make_query(should_terms={'_id': ids}, consistent_order=False)
```

Comment:
Since we're operating asynchronously, I'd be inclined to do this in batches anyway.  It wouldn't be too difficult just to break down the ids into groups of, say, 1000 and churn through them.  A "terms" query takes a max of 1024, so it wouldn't be surprising if a should query has similar limits.